nexus/db-queries/src/db/datastore/probe.rs

```diff
@@ -317,7 +300,200 @@ impl super::DataStore {
+
+        // NOTE: This paginator _must_ use ascending order. See
+        // `list_probe_create_params_for_sled` for details.
+        let mut paginator = Paginator::new(
+            SQL_BATCH_SIZE,
+            dropshot::PaginationOrder::Ascending,
+        );
+        let mut all_probes_for_sled = Vec::new();
+        while let Some(p) = paginator.next() {
+            let mut batch = self
+                .list_probe_create_params_for_sled(
+                    opctx,
+                    sled_id,
+                    &p.current_pagparams(),
+                )
+                .await?;
+            paginator = p.found_batch(&batch, &|probe| probe.id);
+            all_probes_for_sled.append(&mut batch);
+        }
+        Ok(all_probes_for_sled)
+    }
+
+    // Construct a page of probe-create parameters for a specific sled.
+    //
+    // # Panics
+    //
+    // This panics if the `pagparams` is not in ascending order. This method
+    // paginates a big join of all the tables needed to build these parameters.
+    // We can't use the existing `paginated` method for that.
+    async fn list_probe_create_params_for_sled(
+        &self,
+        opctx: &OpContext,
+        sled_id: SledUuid,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<ProbeCreate> {
+        use nexus_db_schema::schema::{
+            external_ip, network_interface, probe, vpc, vpc_subnet,
+        };
+
+        /*
+        assert_eq!(pagparams.direction, dropshot::PaginationOrder::Ascending);
+        let marker = pagparams.marker.cloned().unwrap_or_default();
+        */
+
+        // TODO-correctness: This inner join below assumes exactly one external
+        // IP for each probe. That's true today because of how we specify the
+        // create-params in the external API. That has one IP Pool, and we
+        // create exactly one external IP for the probe from that pool. If we
+        // allow multiple IPs, then we'll need to change this to add group the
+        // external IPs by the probe ID.
+        //
+        // In that case, we'd get rows like:
+        //
+        // probe_id, vpc stuff, vpc_subnet stuff, [eip0, eip1, ...]
+        paginated(probe::dsl::probe, probe::dsl::id, pagparams)
+            .inner_join(
+                external_ip::dsl::external_ip
+                    .on(external_ip::dsl::parent_id
+                        .eq(probe::dsl::id.nullable())),
+            )
+            .inner_join(
+                network_interface::dsl::network_interface
+                    .on(network_interface::dsl::parent_id.eq(probe::dsl::id)),
+            )
+            .inner_join(
+                vpc_subnet::dsl::vpc_subnet
+                    .on(vpc_subnet::dsl::id
+                        .eq(network_interface::dsl::subnet_id)),
+            )
+            .inner_join(
+                vpc::dsl::vpc.on(vpc::dsl::id.eq(vpc_subnet::dsl::vpc_id)),
+            )
+            //.filter(probe::dsl::id.gt(marker))
```

Comment:
remove commented out filter/order/limit?

Reply:
Done in 1bc7af662